src/RazorSdk/Tool/ServerProtocol/ServerLogger.cs

```diff
@@ -124,23 +125,11 @@ public static void Log(string message)
             }
         }
 
-        private static int GetCurrentProcessId()
-        {
-            var process = Process.GetCurrentProcess();
-            return process.Id;
-        }
-
-        private static int GetCurrentThreadId()
-        {
-            return Environment.CurrentManagedThreadId;
-        }
-
-        /// <summary>
-        /// Get the string that prefixes all log entries. Shows the process, thread, and time.
-        /// </summary>
-        private static string GetLoggingPrefix()
-        {
-            return string.Format(CultureInfo.InvariantCulture, "{0} PID={1} TID={2} Ticks={3}: ", s_prefix, GetCurrentProcessId(), GetCurrentThreadId(), Environment.TickCount);
-        }
+        private static int GetCurrentProcessId() =>
+#if NET5_0_OR_GREATER
```

Comment:
Why is the `#if` needed in this file, but not for other files in this project? Doesn't this project only build for .NET 6+?

Reply:
It is actually `Microsoft.NET.Sdk.Razor.Tasks.csproj ` which explicitly includes ServerProtocol and it builds for net472:
https://github.com/dotnet/sdk/blob/45e032c4d0f3625fb5bad6e50067a06b6b7fdc0a/src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj#L35
https://github.com/dotnet/sdk/blob/45e032c4d0f3625fb5bad6e50067a06b6b7fdc0a/src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj#L69